BlazarService/src/main/java/com/hubspot/blazar/util/ModuleBuildLauncher.java

```diff
@@ -108,8 +109,10 @@ private BuildConfig mergeConfig(BuildConfig primary, BuildConfig secondary) {
     Map<String, StepActivationCriteria> stepActivation = new LinkedHashMap<>();
     stepActivation.putAll(secondary.getStepActivation());
     stepActivation.putAll(primary.getStepActivation());
+    Set<String> depends = primary.getDepends().isEmpty() ? secondary.getDepends() : primary.getDepends();
+    Set<String> provides = primary.getProvides().isEmpty() ? secondary.getProvides() : primary.getProvides();
```

Comment:
I didn't merge these because we're currently not merging any lists.
We might want to merge these though because setting up a Java project's dependency on something external would be a huge PITA.